### PR TITLE
Public Cloud: softreboot fix and ssh log upload

### PR DESCRIPTION
**Pull request by pdostal**
- **PC softreboot: Remove stale SSH master processes**
- **PC: Upload ssh_sut log in cleanup**

- Related ticket: [poo#156718](https://progress.opensuse.org/issues/156718)
- Verification run:
    sle-12-SP5-EC2-BYOS-Updates.x86_64	[publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6099)	[22 1](https://pdostal-server.suse.cz/tests/6099)
    sle-15-SP3-EC2-BYOS-Updates.x86_64	[publiccloud_containers@64bit](https://pdostal-server.suse.cz/tests/6097)	[28 1](https://pdostal-server.suse.cz/tests/6097)
    sle-15-SP4-GCE-BYOS-Updates.aarch64	[publiccloud_consoletests@64bit](https://pdostal-server.suse.cz/tests/6100)	[23](https://pdostal-server.suse.cz/tests/6100)
    sle-15-SP4-EC2-CHOST-BYOS-Updates.aarch64	[publiccloud_containers@64bit](https://pdostal-server.suse.cz/tests/6098)	[28 1](https://pdostal-server.suse.cz/tests/6098)
    sle-15-SP6-Azure-BYOS-gen1.x86_64	[publiccloud_consoletests@az_Standard_B2s](https://pdostal-server.suse.cz/tests/6096)	[14](https://pdostal-server.suse.cz/tests/6096)
    sle-15-SP3-EC2-BYOS-Updates.aarch64	[publiccloud_cloud_netconfig@64bit](https://pdostal-server.suse.cz/tests/6095)	[7](https://pdostal-server.suse.cz/tests/6095)
    sle-15-SP5-GCE-Updates.aarch64	[publiccloud_img_proof@64bit](https://pdostal-server.suse.cz/tests/6094)	[28 1](https://pdostal-server.suse.cz/tests/6094)

### CODE REVIEW

File: lib/publiccloud/instance.pm [hunk 1]
```diff
@@ -481,6 +481,9 @@ sub softreboot {
     if ($tunneled) {
         select_console('tunnel-console', await_console => 0);
         ssh_interactive_leave();
+        while (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) == 0) {
```
Comment (grisu48) [inline lib/publiccloud/instance.pm:484]: Is a loop really needed here? Wouldn't this do the same?

```
script_run(sprintf('ssh -O exit %s@%s || true', $args{username}, $self->public_ip))
```

**Review by grisu48 — COMMENTED** [review]
Just a minor suggestion

**Review by ricardobranco777 — APPROVED** [review]

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: lib/publiccloud/instance.pm (hunk 1 above)
Comment (grisu48) [reply lib/publiccloud/instance.pm:484]: I see. In that case I would add a sleep after the `-O exit` call to avoid unnecessary load on the system.

Comment (pdostal) [reply lib/publiccloud/instance.pm]: Done now.

File: lib/publiccloud/instance.pm [hunk 2]
```diff
@@ -481,6 +481,12 @@ sub softreboot {
     if ($tunneled) {
         select_console('tunnel-console', await_console => 0);
         ssh_interactive_leave();
+        my $i = 5;
+        while (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) == 0 && $i != 0) {
+            script_run(sprintf('ssh -O exit %s@%s', $args{username}, $self->public_ip));
+            $i = $i - 1;
+            sleep 5;
+        }
```
Comment (ricardobranco777) [inline lib/publiccloud/instance.pm:489]: It seems more natural to me to just do a `for` loop here, but you may skip this suggestion.

```suggestion
        for (1 .. 5) {
            last if (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) != 0);
            script_run(sprintf('ssh -O exit %s@%s', $args{username}, $self->public_ip));
            sleep 5;
        }
```

Comment (asmorodskyi) [reply lib/publiccloud/instance.pm:489]: agree with Ricardo "for loop" looks better here

**Review by ricardobranco777 — APPROVED** [review]

File: lib/publiccloud/instance.pm [hunk 3]
```diff
@@ -147,7 +147,7 @@ sub _prepare_ssh_cmd {
         $cmd = "\$'$cmd'";
     }
 
-    my $ssh_cmd = sprintf('ssh -t %s "%s@%s" -- %s', $args{ssh_opts}, $args{username}, $self->public_ip, $cmd);
+    my $ssh_cmd = sprintf('ssh -tvE /var/tmp/ssh_sut.log %s "%s@%s" -- %s', $args{ssh_opts}, $args{username}, $self->public_ip, $cmd);
```
Comment (mpagot) [inline lib/publiccloud/instance.pm:150]: this one is not considering someone could specify `-E something` through `ssh_opts` like sles4sap is  doing in:
-  https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20169/files#diff-5404750314cf685b060d425b85a04de2cdfb000225014d52ba83700abe58b2bdR142
- evolved to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20274/files#diff-5404750314cf685b060d425b85a04de2cdfb000225014d52ba83700abe58b2bdR125

Result is that in some situations the result could have `-E something` added twice to the command line 

http://openqaworker15.qa.suse.cz/tests/307727/logfile?filename=serial_terminal.txt

Comment (pdostal) [reply lib/publiccloud/instance.pm:150]: That is true.

**Pull request merged**